docs/documentation.md

```diff
@@ -0,0 +1,430 @@
+
+    ***Note:*** We recommend maintaining <= 50(TBD) duplicate listings of any given NFT.
+
+2. ***Unsupported receiver capability*** A common pitfall during the purchase of an NFT is if `saleCut` receivers don’t have a supported receiver capability because that entitled sale cut would transfer to first valid sale cut receiver. To mitigate this we recommend using the generic receiver from the [`FungibleTokenSwitchboard`](https://github.com/onflow/flow-ft/blob/master/contracts/FungibleTokenSwitchboard.cdc) contract, adding capabilities to support whichever token types the beneficiary wishes to receive. 
+
+## Enabling creator royalties for NFTs
+
+The `NFTStorefrontV2` contract optionally supports paying royalties to the minter account for secondary resales of a NFT. Providing that a seller's NFT supports the [Royalty Metadata View](https://github.com/onflow/flow-nft/blob/21c254438910c8a4b5843beda3df20e4e2559625/contracts/MetadataViews.cdc#L335), the contract dynamically calculates royalties due at the time of listing creation and applies it as a `saleCut` at time of sale. 
+
+```cadence
+// Check whether the NFT implements the MetadataResolver or not.
+if nft.getViews().contains(Type<MetadataViews.Royalties>()) {
+		// Resolve the royalty view
+    let royaltiesRef = nft.resolveView(Type<MetadataViews.Royalties>())?? panic("Unable to retrieve the royalties")
+	  // Fetch the royalties.
+		let royalties = (royaltiesRef as! MetadataViews.Royalties).getRoyalties()
+		// Append the royalties as the salecut
+    for royalty in royalties {
+        self.saleCuts.append(NFTStorefrontV2.SaleCut(receiver: royalty.receiver, amount: royalty.cut * effectiveSaleItemPrice))
+        totalRoyaltyCut = totalRoyaltyCut + royalty.cut * effectiveSaleItemPrice
+    }
+}
+```
+
+Complete transaction available [here](https://github.com/onflow/nft-storefront/blob/main/transactions/sell_item.cdc).
+
+`saleCut` only supports a single token receiver type and therefore beneficiaries of a `saleCut` can only receive the token type used for the purchase. To support different token types for saleCuts we recommend using the [FungibleTokenSwitchboard](https://github.com/onflow/flow-ft/blob/master/contracts/FungibleTokenSwitchboard.cdc) contract.
+
+***Note:*** We recommend that marketplaces honor creator royalties across the Flow ecosystem
+
+## Enabling marketplace commissions for NFT sales
+
+`NFTStorefrontV2` enables optional commissions on trades for marketplaces which require it as a condition to list a NFT for sale. Commission & commission receivers are set by the seller during initial listing creation. At time of purchase the commission amount is paid once only to the commission receiver matching the marketplace receiver address which facilitated the sale. When commission is set but no commission receivers are specified, the commission is paid to [TBD].
```

Comment:
@satyamakgec "When commission is set but no commission receivers are specified, the commission is paid to [TBD]." - who gets paid?

Reply:
Anyone can take the commission, as there is no whitelisting for the commission receivers. Technically the receiver provided at the time of purchase function will receive the commission.